src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1388,13 +1408,13 @@ Literal Literal::substr_after(std::string_view const needle, Node::NodeStorage &
     }
 
     auto const s = this->lexical_form();
-    auto const pos = s.view().find(needle);
 
-    if (pos == std::string_view::npos) {
+    const auto r = una::casesens::search_utf8(s.view(), needle);
+
+    if (!r)
         return Literal::make_simple_unchecked("", node_storage);
-    }
 
-    auto const substr = s.view().substr(pos + needle.size());
+    auto substr = static_cast<std::string_view>(s.view()).substr(r.pos() + needle.size());  // search_utf8 returns byte position, not unicode character position
```

Comment:
why the `static_cast` to `std::string_view`?

Reply:
to use `std::string_view::substr` instead of `std::string::substr`, which does not allocate any memory

Reply:
`s.view()` returns a `std::string`?

Reply:
oh, you are correct
that s.view() was a std::string in one of the previous versions and i forgot to remove the cast